tartufo/scanner.py

```diff
@@ -295,13 +296,20 @@ def rule_matches(rule: Rule, string: str, path: str) -> bool:
         :return: True if string and path matched, False otherwise.
         """
         match = False
-        if rule.pattern:
-            match = rule.pattern.search(string) is not None
-        if rule.path_pattern:
-            match = match and rule.path_pattern.search(path) is not None
+        if rule.re_match_type == "match":
+            if rule.pattern:
+                match = rule.pattern.match(string) is not None
+            if rule.path_pattern:
+                match = match and rule.path_pattern.match(path) is not None
```

Comment:
Because `rule.pattern` is mandatory and `rule.path_pattern` is always a `re` (if you took the advice above), I think this can be simplified:
```suggestion
            match = rule.pattern.match(string) is not None and rule.path_pattern.match(path) is not None
```
or presumably even `match = bool(rule.pattern.match(string) and rule.path_pattern.match(path))`